language/tools/move-package/src/package_lock.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use named_lock::{NamedLock, NamedLockGuard};
+use once_cell::sync::Lazy;
+use std::sync::{Mutex, MutexGuard};
+
+const PACKAGE_LOCK_NAME: &str = "move_pkg_lock";
+static PACKAGE_THREAD_MUTEX: Lazy<Mutex<()>> = Lazy::new(|| Mutex::new(()));
+static PACKAGE_PROCESS_MUTEX: Lazy<NamedLock> =
+    Lazy::new(|| NamedLock::create(PACKAGE_LOCK_NAME).unwrap());
+
+// The package lock is a lock held across threads and processes
```

Comment:
Probably useful to explain what the package lock is protecting in some detail (I think the global package cache and its subdirectories) + when it needs to be acquired.